internal/tiltfile/config/config.go

```diff
@@ -31,6 +39,19 @@ func (e *Extension) NewState() interface{} {
 
 var _ starkit.StatefulExtension = &Extension{}
 
+func (e *Extension) OnExec(t *starlark.Thread, path string) error {
+	dir := filepath.Dir(path)
```

Comment:
so...every time you sub-include a Tiltfile, it creates a new tilt_config.json in that Tiltfile's directory and overwrites the current UserConfigState.

are you sure that's the behavior you want? can you add a test for this?

I would have expected this to be an OnStart extension that only affects the main Tiltfile load, OnExec extensions are pretty rare

Reply:
> are you sure that's the behavior you want?

Nope, I failed to account for loads/includes in this PR. Thanks. This interaction should be tested regardless of what the behavior is.

> I would have expected this to be an OnStart extension that only affects the main Tiltfile load, OnExec extensions are pretty rare

If we want Tiltfiles to be composable, I think we'll need this to be an OnExec extension so that we can get each config path separately (or maybe a thread-local that tells us the current tiltfile path), but yeah ok I'll find another way to get the path.

> so...every time you sub-include a Tiltfile, it creates a new tilt_config.json in that Tiltfile's directory and overwrites the current UserConfigState.

(FWIW, just to make sure we're on the same page, the intent is that Tilt is no longer creating tilt_config.json)

Reply:
re: thread-local that tells us the current tiltfile path: https://github.com/windmilleng/tilt/blob/master/internal/tiltfile/starkit/path.go#L18